# **INSERT PROJECT TITLE HERE**
##CS-280 Introduction to Data Science Final Project
Contributors: Mohammed Asad | Nathaniel Gordon | Brendan Hasara





## **Purpose**
The purpose of this project is to achieve a deeper and more meaningful understanding of data manipulation, visualization, and staistical analysis.

## **The Questions That Needs Answering**

When looking at this dataset, we came to an overarching question: How do the variations in socioeconomic status influence the perception of trust and happiness throughout different countries?

To answer this, we needed to ask a few deeoer questions to paint a more in-depth understanding of the overarching question at hand.



1. Does the average GDP per capita have a direct relationship with the perceived happiness (life ladder) of that country's population?
2. Is there an overall significant cause/effect relationship between social support and generosity?
3. Does freedom to make life choices influence the confidence placed in each country's national government?
4. Is there a need for a higher GDP per capita in order to expect a higher health life expectancy at birth?
5. On a global scale, are people more happy or sad over the years?



## **Important Terminology**

...

### NOTE TO STUDENTS

Delete this code cell at the end.  

* EVERY code cell must have a comment in it at the top explaining what is in the code cell.

* There should be normal coding comments within the code cell too, especially if there is anything unexpected or not obvious happening in the code.

* NOOOOOOOOOO code cell should include any interpretations of the data.  Those remarks should be in markdown cells  directly before or after the code cells.

**Please delete the bolded comments from Dr. R before submission.**

In [1]:
# ALL imports here
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

---
#### CS 280 Final Project - FA23
---

# Title (Give a title, based on your topic. Also be sure to rename the file with your initials.)

### By: Mo Asad, Nathaniel Gordon, Brendan Hasara


---

## 1. Introduction

For this project, our group explored... (This section will be finished at a later date and can be skipped at this time.)

---

## 2. Data

A direct link to the data can be found at the following website.
 * https://data.world/laurel/world-happiness-report-data/workspace/file?filename=online-data-chapter-2-whr-2017.xlsx

**This World Happiness Report was collected by all countries in the world from the years of 2005 to 2016. The report measured countries on a point system for various qualities of life including life ladder, social support, trustworthiness of others, life expectancy, perceptions of corruptions, etc.**

In [2]:
#importing the data from github
world = pd.read_csv("https://raw.githubusercontent.com/GoodGuyMomo/CS-280-Final-Project/main/happy_data.csv")
world

WP5 Country      country  year  Life Ladder  Log GDP per capita  \
0     Afghanistan  Afghanistan  2008     3.723590            7.197130   
1     Afghanistan  Afghanistan  2009     4.401778            7.362664   
2     Afghanistan  Afghanistan  2010     4.758381            7.416260   
3     Afghanistan  Afghanistan  2011     3.831719            7.445761   
4     Afghanistan  Afghanistan  2012     3.782938            7.549241   
...           ...          ...   ...          ...                 ...   
1415     Zimbabwe     Zimbabwe  2012     4.955101            7.407775   
1416     Zimbabwe     Zimbabwe  2013     4.690188            7.429061   
1417     Zimbabwe     Zimbabwe  2014     4.184451            7.443748   
1418     Zimbabwe     Zimbabwe  2015     3.703191            7.431285   
1419     Zimbabwe     Zimbabwe  2016     3.735400            7.422308   

      Social support  Healthy life expectancy at birth  \
0           0.450662                         47.550438   
1           0.552308                         47.859673   
2           0.539075                         48.159512   
3           0.521104                         48.451160   
4           0.520637                         48.738346   
...              ...                               ...   
1415        0.896476                         45.319149   
1416        0.799274                         47.000294   
1417        0.765839                         48.576164   
1418        0.735800                         50.152035   
1419        0.768425                         51.727909   

      Freedom to make life choices  Generosity  Perceptions of corruption  \
0                         0.718114    0.183062                   0.881686   
1                         0.678896    0.204633                   0.850035   
2                         0.600127    0.138529                   0.706766   
3                         0.495901    0.176108                   0.731109   
4                         0.530935    0.247713                   0.775620   
...                            ...         ...                        ...   
1415                      0.469531   -0.059864                   0.858691   
1416                      0.575884   -0.062995                   0.830937   
1417                      0.642034   -0.033504                   0.820217   
1418                      0.667193   -0.081744                   0.810457   
1419                      0.732971   -0.053070                   0.723612   

      ...  GINI index (World Bank estimate)  \
0     ...                               NaN   
1     ...                               NaN   
2     ...                               NaN   
3     ...                               NaN   
4     ...                               NaN   
...   ...                               ...   
1415  ...                               NaN   
1416  ...                               NaN   
1417  ...                               NaN   
1418  ...                               NaN   
1419  ...                               NaN   

      GINI index (World Bank estimate), average 2000-13  \
0                                                   NaN   
1                                                   NaN   
2                                                   NaN   
3                                                   NaN   
4                                                   NaN   
...                                                 ...   
1415                                                NaN   
1416                                                NaN   
1417                                                NaN   
1418                                                NaN   
1419                                                NaN   

      gini of household income reported in Gallup, by wp5-year  \
0                                                   NaN          
1                                              0.441906          
2                                  

### 2.1 Summary of Variables within the XXX Table

In [3]:
# List of columns to be removed
columns_to_remove = ['WP5 Country',
                     'GINI index (World Bank estimate)', 
                     'GINI index (World Bank estimate), average 2000-13', 
                     'gini of household income reported in Gallup, by wp5-year', 
                     'Most people can be trusted, WVS round 1981-1984', 
                     'Most people can be trusted, WVS round 1989-1993', 
                     'Most people can be trusted, WVS round 1994-1998', 
                     'Most people can be trusted, WVS round 1999-2004',
                     'Most people can be trusted, WVS round 2005-2009',
                     'Most people can be trusted, WVS round 2010-2014']

# Remove multiple columns from the DataFrame
new_world = world.drop(columns=columns_to_remove)

#Rename some columns for clarity
new_world.rename(columns={'Log GDP per capita': 'GDP per capita',
                         'Most people can be trusted, Gallup': 'Trustworthiness of others'}, inplace = True)
new_world

country  year  Life Ladder  GDP per capita  Social support  \
0     Afghanistan  2008     3.723590        7.197130        0.450662   
1     Afghanistan  2009     4.401778        7.362664        0.552308   
2     Afghanistan  2010     4.758381        7.416260        0.539075   
3     Afghanistan  2011     3.831719        7.445761        0.521104   
4     Afghanistan  2012     3.782938        7.549241        0.520637   
...           ...   ...          ...             ...             ...   
1415     Zimbabwe  2012     4.955101        7.407775        0.896476   
1416     Zimbabwe  2013     4.690188        7.429061        0.799274   
1417     Zimbabwe  2014     4.184451        7.443748        0.765839   
1418     Zimbabwe  2015     3.703191        7.431285        0.735800   
1419     Zimbabwe  2016     3.735400        7.422308        0.768425   

      Healthy life expectancy at birth  Freedom to make life choices  \
0                            47.550438                      0.718114   
1                            47.859673                      0.678896   
2                            48.159512                      0.600127   
3                            48.451160                      0.495901   
4                            48.738346                      0.530935   
...                                ...                           ...   
1415                         45.319149                      0.469531   
1416                         47.000294                      0.575884   
1417                         48.576164                      0.642034   
1418                         50.152035                      0.667193   
1419                         51.727909                      0.732971   

      Generosity  Perceptions of corruption  Positive affect  Negative affect  \
0       0.183062                   0.881686         0.517637         0.258195   
1       0.204633                   0.850035         0.583926         0.237092   
2       0.138529                   0.706766         0.618265         0.275324   
3       0.176108                   0.731109         0.611387         0.267175   
4       0.247713                   0.775620         0.710385         0.267919   
...          ...                        ...              ...              ...   
1415   -0.059864                   0.858691         0.669279         0.177311   
1416   -0.062995                   0.830937         0.711885         0.182288   
1417   -0.033504                   0.820217         0.725214         0.239111   
1418   -0.081744                   0.810457         0.715079         0.178861   
1419   -0.053070                   0.723612         0.737636         0.208555   

      Confidence in national government  Democratic Quality  Delivery Quality  \
0                              0.612072           -1.962724         -1.665877   
1                              0.611545           -2.082451         -1.647945   
2                              0.299357           -2.015700         -1.630097   
3                              0.307386           -1.943686         -1.620443   
4                              0.435440           -1.855898         -1.435541   
...                                 ...                 ...               ...   
1415                           0.407084           -1.117682         -1.477044   
1416                           0.527755           -1.031373         -1.471021   
1417                           0.566209           -0.962947         -1.468772   
1418                           0.590012                 NaN               NaN   
1419                           0.699344                 NaN               NaN   

      Standard deviation of ladder by country-year  \
0                                         1.774662   
1                                         1.722688   
2                                         1.878622   
3                                         1.785360   
4                                         1.798283   
...       

This code is to count the number of *missing* values for each variable/column in the dataframe.

In [4]:
#To count the number of missing values for each variable/column
nan_counts = new_world.isna().sum()
print(nan_counts)

country                                                 0
year                                                    0
Life Ladder                                             0
GDP per capita                                         35
Social support                                         12
Healthy life expectancy at birth                        9
Freedom to make life choices                           34
Generosity                                             92
Perceptions of corruption                              84
Positive affect                                        17
Negative affect                                        11
Confidence in national government                     195
Democratic Quality                                    292
Delivery Quality                                      291
Standard deviation of ladder by country-year            0
Standard deviation/Mean of ladder by country-year       0
Trustworthiness of others                            1240
dtype: int64


This code is to count the number of *unique* values for each variable/column in the dataframe.

In [5]:
# Count unique values for each column
unique_counts = new_world.nunique()
print(unique_counts)

country                                               164
year                                                   12
Life Ladder                                          1420
GDP per capita                                       1385
Social support                                       1408
Healthy life expectancy at birth                     1366
Freedom to make life choices                         1386
Generosity                                           1328
Perceptions of corruption                            1336
Positive affect                                      1403
Negative affect                                      1409
Confidence in national government                    1225
Democratic Quality                                   1128
Delivery Quality                                     1129
Standard deviation of ladder by country-year         1420
Standard deviation/Mean of ladder by country-year    1419
Trustworthiness of others                             180
dtype: int64


### 2.2 Code used to Summarize Variables

In this section we include the code used to describe the variables in the next section.  

In [6]:
#Get descriptive stats for all columns in the dataframe
new_world.describe()

year  Life Ladder  GDP per capita  Social support  \
count  1420.000000  1420.000000     1385.000000     1408.000000   
mean   2011.303521     5.428571        9.203860        0.811142   
std       3.149570     1.121719        1.183485        0.119286   
min    2005.000000     2.687553        6.354652        0.290184   
25%    2009.000000     4.600384        8.306463        0.750623   
50%    2011.000000     5.317440        9.378672        0.833718   
75%    2014.000000     6.269866       10.159828        0.904155   
max    2016.000000     8.018934       11.815410        0.987343   

       Healthy life expectancy at birth  Freedom to make life choices  \
count                       1411.000000                   1386.000000   
mean                          61.965278                      0.723822   
std                            8.198405                      0.146490   
min                           36.167667                      0.257534   
25%                           57.029228                      0.629931   
50%                           63.782501                      0.739609   
75%                           67.936405                      0.838298   
max                           76.408699                      0.983803   

        Generosity  Perceptions of corruption  Positive affect  \
count  1328.000000                1336.000000      1403.000000   
mean      0.000201                   0.755678         0.709873   
std       0.159385                   0.186446         0.107631   
min      -0.324367                   0.035198         0.362498   
25%      -0.110767                   0.698339         0.622200   
50%      -0.017912                   0.810556         0.717980   
75%       0.092981                   0.882902         0.802945   
max       0.542249                   0.983276         0.943621   

       Negative affect  Confidence in national government  Democratic Quality  \
count      1409.000000                        1225.000000         1128.000000   
mean          0.260295                           0.464820           -0.143355   
std           0.082701                           0.178399            0.865859   
min           0.083426                           0.068769           -2.310710   
25%           0.202129                           0.328487           -0.800332   
50%           0.249856                           0.453570           -0.252632   
75%           0.307471                           0.592506            0.649054   
max           0.704590                           0.979501            1.538103   

       Delivery Quality  Standard deviation of ladder by country-year  \
count       1129.000000                                   1420.000000   
mean          -0.002564                                      1.983658   
std            0.961750                                      0.366721   
min           -2.167530                                      0.863034   
25%           -0.705457                                      1.732018   
50%           -0.227479                                      1.948063   
75%            0.671737                                      2.191734   
max            2.200550                                      3.527820   

       Standard deviation/Mean of ladder by country-year  \
count                                        1420.000000   
mean                                            0.383186   
std                                             0.114287   
min                                             0.133908   
25%                                             0.309646   
50%                                             0.367580   
75%                                             0.445621   
max                                             1.022769   

       Trustworthiness of others  
count                 180.000000  
mean                    0.226295  
std                     0.119079  
min                     0.066618  
25%                     0.139773  
50%               

The "new_world" table has 17 columns/variables and 1420 rows. The original dataset had 27 columns, but we decided to remove certain columns either because they were redundant or irrelavent to the the purpose of this study. We will now explain each variable, giving their units, indicating their types, and providing some descriptive statistics if necessary.

**There are 2 categorical variables**
        
* **country** – Country name
    * Number of missing values: 0
    * Contains 164 distinct values: 
        *   Afghanistan, Albania, Algeria, Angola, Argentina, Armenia, Australia, Austria, Azerbaijan, Bahrain

* **year** – Year
    * Number of missing values: 0
    * Contains 12 distinct values: 
        *   2005-2016
        
**There are 15 numerical variables**

* **Life ladder** – score that represents one's own position in life with 10 being the best and 1 being the worst
    * Number of missing values: 0
    * max value: 8.018934
    * min value: 2.687553
    * average value: 5.428571

* **GDP per capita** – GDP per capita in purchasing power parity at constant 2011 int. dollar prices
    * Number of missing values: 35
    * max value: 11.815410
    * min value: 6.354652
    * average value: 9.203860
    
* **Social support** – national average of a binary response (between 0 and 1) for having support on others for when times are tough
    * Number of missing values: 12
    * max value: 0.987343
    * min value: 0.290184
    * average value: 0.811142

* **Healthy life expectancy at birth** – The time series of healthy life expectancy at birth
    * Number of missing values: 9
    * max value: 76.408699
    * min value: 36.167667
    * average value: 61.965278

* **Freedom to make life choices** – National avg responses to “Are you satisfied or dissatisfied with your freedom to choose what you do with your life?" (Binary; yes or no; 0-1)
    * Number of missing values: 34
    * max value: 0.983803
    * min value: 0.257534
    * average value: 0.723822

* **Generosity** – National avg responses to “Have you donated money to a charity in the past month?” on GDP per capita (yes or no)

    * Number of missing values: 92
    * max value: 0.542249
    * min value: -0.324367
    * average value: 0.000201
    
* **Perceptions of corruption** – Avg response to “Is corruption widespread throughout the government, businesses, etc.” (yes or no)
    * Number of missing values: 84
    * max value: 0.983276
    * min value: 0.035198
    * average value: 0.755678
    
* **Positive affect** – The average of three positive affect measures in GWP: happiness, laugh and enjoyment in the Gallup World Poll (yes or no)
    * Number of missing values: 17
    * max value: 0.943621
    * min value: 0.362498
    * average value: 0.709873
    
* **Negative affect** – average of three negative affect measures in GWP: worry, sadness, and anger (yes or no)
    * Number of missing values: 11
    * max value: 0.704590
    * min value: 0.083426
    * average value: 0.260295
    
* **Confidence in national government** – yes or no response to the question "Do you have confidence in your national government?"
    * Number of missing values: 195
    * max value: 0.979501
    * min value: 0.068769
    * average value: 0.464820    

* **Democratic quality** – 0-1 rating based on six dimensions: Voice and Accountability, Political Stability and Absence of Violence, Government Effectiveness, Regulatory Quality, Rule of Law, Control of Corruption.
    * Number of missing values: 292
    * max value: 1.538103
    * min value: -2.310710
    * average value: -0.143355
    
* **Delivery Quality** – similar to Democratic quality, based on the six dimensions previously mentioned on a 0-1 scale
    * Number of missing values: 291
    * max value: 2.200550
    * min value: -2.167530
    * average value: -0.002564   

* **Standard deviation of ladder by country-year** – standard deviation of life ladder per country and year
    * Number of missing values: 0
    * max value: 3.527820
    * min value: 0.863034
    * average value: 1.983658  

* **Standard deviation/mean of ladder by country-year** – standard deviation and mean of life ladder per country and year
    * Number of missing values: 0
    * max value: 1.022769
    * min value: 0.133908
    * average value: 0.383186  

* **Trustworthiness of others** – Avg response to “Generally speaking, would you say that most people can be trusted or that you have to be careful..."
    * Number of missing values: 1240
    * max value: 0.640332
    * min value: 0.066618
    * average value: 0.226295  

---

## Playground

In [8]:
# Vis 1

What you learned

In [9]:
# Vis 2

What you learned

In [10]:
# Vis 3

What you learned

---
## 3. Results

You're welcome to organize your final file as needed.  You will certainly need a results section and a conclusion section.

---

## 4. Conclusion